staging/src/k8s.io/kubectl/pkg/util/completion/completion.go

```diff
@@ -112,6 +115,40 @@ func PodResourceNameAndContainerCompletionFunc(f cmdutil.Factory) func(*cobra.Co
 	}
 }
 
+// SetImageCompletionFunc a function to complete for kubectl set image.
+func SetImageCompletionFunc(f cmdutil.Factory) func(*cobra.Command, []string, string) ([]string, cobra.ShellCompDirective) {
```

Comment:
Could you please use `restclientgetter` instead of tight coupling to `cmdutil.Factory`?

Reply:
Oops, i didn't know about that. Current completion code is heavily coupled to `cmdutil.Factory` maybe later we can update this file. 
Thanks i'll fix this.

Reply:
Checked the code little bit, bottom line we reach to these lines
https://github.com/kubernetes/kubernetes/blob/1cbe7b6b439463623f461a4be96fb5e13fa2525c/staging/src/k8s.io/kubectl/pkg/cmd/get/get.go#L441

https://github.com/kubernetes/kubernetes/blob/1cbe7b6b439463623f461a4be96fb5e13fa2525c/staging/src/k8s.io/kubectl/pkg/cmd/get/get.go#L458

which is using `cmdutil.Factory`. I'm not sure if `restclientgetter` can do the job here or not, please correct me if i'm wrong.